sgx/enclave/src/lib.rs

```diff
@@ -105,3 +110,106 @@ fn wasm(
     copy_string_to_cstr_ptr(&value, result_ptr, result_capacity, result_len)?;
     Ok(())
 }
+
+#[no_mangle]
+pub extern "C" fn sgx_multiply(
+    adapter_str_ptr: *const u8,
+    adapter_str_len: usize,
+    input_str_ptr: *const u8,
+    input_str_len: usize,
+    output_bin: *mut u8,
+    output_max_len: usize,
+) -> sgx_status_t {
```

Comment:
I think this is confusing things, we should probably replace this with our own error type.